python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -316,8 +316,10 @@ def _get_device(device: str) -> str:
        or type(device) == int
        or device is None):
         if device == Device.AUTO.value:
-            return (Device.CUDA.value if torch.cuda.is_available()
-                    else Device.CPU.value)
+            if torch.cuda.is_available():
```

Comment:
did you have a chance to look into why the tests did not catch this - and is it possible to write a test for this?